sourcegraph/sourcegraph.proto

```diff
@@ -288,58 +289,115 @@ message RepoList {
 
 service RepoBadges {
 	// ListBadges lists the available badges for repo.
-	rpc ListBadges(RepoSpec) returns (BadgeList);
+	rpc ListBadges(RepoSpec) returns (BadgeList) {
+		option (google.api.http) = {
+			get: "/v1/repo_badges/list_badges"
+		};
+	};
 
 	// ListCounters lists the available counters for repo.
-	rpc ListCounters(RepoSpec) returns (CounterList);
+	rpc ListCounters(RepoSpec) returns (CounterList) {
+		option (google.api.http) = {
+			get: "/v1/repo_badges/list_counters"
+		};
+	};
 
 	// RecordHit records a visit to a repo (that will be reflected in
 	// its counter).
-	rpc RecordHit(RepoSpec) returns (pbtypes.Void);
+	rpc RecordHit(RepoSpec) returns (pbtypes.Void) {
+		option (google.api.http) = {
+			put: "/v1/repo_badges/record_hit"
+		};
+	};
 
 	// CountHits returns the hit count (optionally in a recent time
 	// period).
-	rpc CountHits(RepoBadgesCountHitsOp) returns (RepoBadgesCountHitsResult);
+	rpc CountHits(RepoBadgesCountHitsOp) returns (RepoBadgesCountHitsResult) {
+		option (google.api.http) = {
+			get: "/v1/repo_badges/count_hits"
+		};
+	};
 }
 
 service RepoStatuses {
 	// GetCombined fetches the combined repository status for the given commit.
-	rpc GetCombined(RepoRevSpec) returns (CombinedStatus);
+	rpc GetCombined(RepoRevSpec) returns (CombinedStatus) {
+		option (google.api.http) = {
+			get: "/v1/repo_statuses/get_combined"
+		};
+	};
 
 	// Create creates a repository status for the given commit.
-	rpc Create(RepoStatusesCreateOp) returns (RepoStatus);
+	rpc Create(RepoStatusesCreateOp) returns (RepoStatus) {
+		option (google.api.http) = {
+			post: "/v1/repo_statuses/post"
+		};
+	};
 }
 
 // Repos exposes information about and actions on both locally hosted
 // and remote repositories.
 service Repos {
 	// Get fetches a repository.
-	rpc Get(RepoSpec) returns (Repo);
+	rpc Get(RepoSpec) returns (Repo) {
+		option (google.api.http) = {
+			get: "/v1/repos/get"
+		};
+	};
 
 	// List repositories.
-	rpc List(RepoListOptions) returns (RepoList);
+	rpc List(RepoListOptions) returns (RepoList) {
+		option (google.api.http) = {
+			get: "/v1/repos/list"
+		};
+	};
 
 	// Create creates a new repository.
-	rpc Create(ReposCreateOp) returns (Repo);
+	rpc Create(ReposCreateOp) returns (Repo) {
+		option (google.api.http) = {
+			post: "/v1/repos/create"
+		};
+	};
 
 	// Delete removes a repository.
-	rpc Delete(RepoSpec) returns (pbtypes.Void);
+	rpc Delete(RepoSpec) returns (pbtypes.Void) {
+		option (google.api.http) = {
+			delete: "/v1/repos/delete"
+		};
+	};
```

Comment:
That helps a little, but I'm not sure if it makes sufficient difference. So, `GET /v1/repos/www.example.com/user/repo` then.

The problem I see is that it may clash with other subpaths we may have, like `GET /v1/repos/list`. Since "repository path" can (and should) be anything, it'd be harder to say, "well, it's everything except these certain things".